src/lib/routes/admin-api/project/api-token.ts

```diff
@@ -157,6 +159,19 @@ export class ProjectApiTokenController extends Controller {
     ): Promise<any> {
         const createToken = await createApiToken.validateAsync(req.body);
         const { projectId } = req.params;
+        const permissionRequired = tokenTypeToCreatePermission(
+            createToken.type,
+        );
+        const hasPermission = await this.accessService.hasPermission(
+            req.user,
+            permissionRequired,
+            projectId,
+        );
```

Comment:
This combines the permission to create project-specific API tokens with the root role permission that determines whether you can create CLIENT or FRONTEND tokens